### PR TITLE
47

### PR DESCRIPTION
**Pull request by johnbender**
Addresses issue #47 using a `paste` binding to fix numeric inputs.

### CODE REVIEW

File: js/numeric-input.js [hunk 1]
```diff
@@ -96,6 +97,30 @@
...
+			pastedText = event.clipboardData.getData('text/plain');
+		}
+
+		// if we were unable to get the pasted text avoid doing anything
+		if( !pastedText ){
+			return;
+		}
+
+		// otherwise force the text to look right
+		this.el.value = pastedText.replace(/[^0-9\.,]*/g, "");
```
Comment (johnbender) [inline js/numeric-input.js]: This regular expression may need to be changed.

File: js/numeric-input.js [hunk 2]
```diff
@@ -28,12 +28,14 @@
 		this.isNavDisabled =
 			(this.$el.attr("data-numeric-input-nav-disabled") !== null &&
 			 this.$el.attr("data-numeric-input-nav-disabled") !== undefined) ||
-			this.$el.hasClass("formcore-disable-spinner");
+			(this.$el.attr("class") || "").indexOf("formcore-disable-spinner") >= 0;
```
Comment (johnbender) [inline js/numeric-input.js]: This prevents a failure when `hasClass` isn't available.

**Comment by johnbender** [comment]
http://47.origin.formcore.fgview.com/views/text-inputs.html

Test there in the CC number with `1234-1234-1234-1234`

**Comment by toddparker** [comment]
@johnbender - I pasted that in and got this validation error. Might be unrelated, but makes me wonder

image

**Comment by johnbender** [comment]
@toddparker 

Try `4111 1111 1111 1111` which is a valid CC number, works fine. Will defer to @zachleat though.

**Comment by toddparker** [comment]
Ah, I see. Ok, worked with `4111-1111-1111-1111`

**Pull request merged**